### PR TITLE
Inactive Members

### PR DESCRIPTION
**Pull request by mienaikoe**
- Adds two endpoints. One for creating an inactive user, and another for moving people between roles
- Modifies the removeMember endpoint to take in multiple people
- Adds a page for Inactive Users, and a link out from the bottom of Roster
  - Can create inactive members and move them to membership

I will address moving people into the inactive member section when I apply checkboxes to the roster in a different PR. (see  https://trello.com/c/y7KDglrn/287-checkbox-table-on-roster)

### CODE REVIEW

File: server/controllers/TeamController.js [hunk 1]
```diff
@@ -205,27 +216,41 @@ const TeamController = {
...
         _id: {
           $in: attendingBoatIds
         }
       },{
         $pull: {
           seats: {
-            person: person_id
+            person: {
+              $in: people_ids,
+            }
```
Comment (k3ntako) [inline server/controllers/TeamController.js]: It didn't seem to remove the member from the boat. Was this working for you?

File: client/models/Team.js [hunk 2]
```diff
@@ -47,12 +60,45 @@ class Team extends Model {
...
+          pulled = this.pullRole("coaches", person);
+        }
+        if( role === 'members' ){
+          this.members.push(person)
+        } else if( !pulled ) {
+          pulled = this.pullRole("members", person);
+        }
+        if( role === 'inactives' ){
+          this.inactives.push(person);
+        }
```
[the remark this answers is not in the document]
Comment (mienaikoe) [reply client/models/Team.js]: Yes. They need to be invited first.

File: client/pages/Inactives/index.js [hunk 3]
```diff
@@ -0,0 +1,159 @@
+import React from 'react';
+import { Link } from 'react-router-dom'
+import CheckboxModelTable from '../../components/Table/CheckboxModelTable';
+import { TableHead, TableRow, TableCell } from 'react-toolbox/lib/table';
```
[the remark this answers is not in the document]
Comment (mienaikoe) [reply client/pages/Inactives/index.js]: Done in roster table PR

File: client/pages/Inactives/index.js [hunk 4]
```diff
@@ -0,0 +1,159 @@
+import React from 'react';
+import { Link } from 'react-router-dom'
+import CheckboxModelTable from '../../components/Table/CheckboxModelTable';
```
[the remark this answers is not in the document]
Comment (mienaikoe) [reply client/pages/Inactives/index.js]: Done in roster table PR

File: client/pages/Team/Roster.js [hunk 5]
```diff
@@ -181,9 +181,14 @@ class Roster extends React.Component {
       }
     ];
 
-    let invites;
+    let invites, inactives;
     if( me.isCoach() ){
       invites = this.renderInvites();
+      inactives = <div className={styles.inactiveLink}>
+        <Link to='/inactives'>
+          View Inactive Members ➤
```
[the remark this answers is not in the document]
Comment (mienaikoe) [reply client/pages/Team/Roster.js]: I'll look for a FontIcon

File: server/controllers/TeamController.js (hunk 1 above)
Comment (mienaikoe) [reply server/controllers/TeamController.js]: Put in some replication steps plz.

File: client/pages/Team/Roster.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (mienaikoe) [reply client/pages/Team/Roster.js]: The FontIcon has issues with alignment w/ the text (probably bcause unicode glyphs are made to flow with text. I'm going to keep it as this arrow.

**Pull request merged**